File: Typescript/src/destCity.ts
export function destCity(paths: string[][]): string {
	let ret = "";
	let start = new Set<string>();
	for (const path of paths) {
		start.add(path[0]);
	}
	for (const path of paths) {
		if (!start.has(path[1])) {
			ret = path[1];
			break;
		}
	}
	return ret;
};

File: Cs/Solution/DestCity.cs
using System.Collections.Generic;

namespace Solution
{
	public partial class Solution
	{
		public string DestCity(IList<IList<string>> paths)
		{
			HashSet<string> set = new();
			foreach (IList<string> path in paths)
			{
				set.Add(path[0]);
			}
			foreach (IList<string> path in paths)
			{
				if (!set.Contains(path[1]))
				{
					return path[1];
				}
			}
			return "";
		}
	}
}
